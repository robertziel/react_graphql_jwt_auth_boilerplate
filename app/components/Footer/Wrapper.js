/* eslint-disable indent */

import styled from 'styled-components';

import { colors } from 'styles/constants';

const Wrapper = styled.footer`
  position: fixed;
  bottom: 0px;
  right: 0px;
  width: ${({ marginLeft }) =>
    marginLeft ? `calc(100% - ${marginLeft}px)` : '100%'};
  height: 40px;
  background-color: #ffffff;
  border-top: rgba(0, 0, 0, 0.02) solid 1px;
  transition: width 100ms ease-out;

  .footer-right {
    float: right;
    height: 40px;
    width: 40px;
    margin-right: 5px;
  }

  .message {
    text-transform: uppercase;
    margin: 8px;
    float: left;

    a {
      color: ${colors.main};
      text-decoration: none;
      font-weight: 600;
    }
  }
`;

export default Wrapper;
