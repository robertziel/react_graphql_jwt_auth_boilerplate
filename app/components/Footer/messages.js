import { defineMessages } from 'react-intl';

export const scope = 'app.components.Footer';

const messages = defineMessages({
  authorMessage: {
    id: `${scope}.author.message`,
    defaultMessage: `
      Designed by {author}.
    `,
  },
});

export default messages;
