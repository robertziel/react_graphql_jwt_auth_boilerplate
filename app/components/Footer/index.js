import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import LanguageToggle from 'components/LanguageToggle/index';

import Wrapper from './Wrapper';
import messages from './messages';

function Footer(props) {
  return (
    <Wrapper marginLeft={props.marginLeft}>
      <section>
        <div className="message">
          <FormattedMessage
            {...messages.authorMessage}
            values={{
              author: (
                <a href="https://robertz.co" target="_blank">
                  Robert Zieli≈Ñski
                </a>
              ),
            }}
          />
        </div>
        <div className="footer-right">
          <LanguageToggle />
        </div>
      </section>
    </Wrapper>
  );
}

Footer.propTypes = {
  marginLeft: PropTypes.number,
};

export default Footer;
