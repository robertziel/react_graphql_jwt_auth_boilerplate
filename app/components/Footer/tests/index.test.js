import React from 'react';
import { IntlProvider } from 'react-intl';

import { mount } from 'enzyme';

import Footer from '../index';

// Mock LanguageToggle required by Footer
/* eslint-disable react/prop-types */
jest.mock('components/LanguageToggle/index', () => () => (
  <div>LanguageToggle</div>
));
/* eslint-enable react/prop-types */

describe('<Footer />', () => {
  it('should render and match the snapshot', () => {
    const wrapper = mount(
      <IntlProvider locale="en">
        <Footer />
      </IntlProvider>,
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
