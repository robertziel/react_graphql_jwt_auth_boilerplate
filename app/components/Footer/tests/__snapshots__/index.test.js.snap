// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and match the snapshot 1`] = `
"<footer class=\\"Wrapper-wcfdfo-0 fvxkdw\\"><section><div class=\\"message\\"><span>
      Designed by <a href=\\"https://robertz.co\\" target=\\"_blank\\">Robert Zieli≈Ñski</a>.
    </span></div><div class=\\"footer-right\\"><div>LanguageToggle</div></div></section></footer>"
`;
